content/python/pagination.py.md

```diff
@@ -1,52 +1,143 @@
+    # paginator.list_servers()
 ```
+
+## [A simple pagination example](#simple-example) {#simple-example .anchor-link}
+
+```python
+
+def list_servers(self):
+    orderBy = {"hardware": {"id": self.DESC}}
+    objectMask = "mask[id,hostname]"
+    servers = self.client.iter_call('SoftLayer_Account', 'getHardware', filter=orderBy, mask=objectMask, limit=10)
+    print("Id, Hostname")
+    for server in servers:
+        print("{}, {}".format(server.get('id'), server.get('hostname')))
+
+```
+
+## [List Users](#list_users) {#list_users .anchor-link}
+
+```python
+def list_users(self):
+    orderBy = {"users": {"id": self.DESC}}
+    objectMask = "mask[id,username]"
+    users = self.client.iter_call('SoftLayer_Account', 'getUsers', filter=orderBy, mask=objectMask, limit=10)
+    print("Id, Username")
+    for user in users:
+        print("{}, {}".format(user.get('id'), user.get('username')))
+```
+
+## [List Global Ips](#list_gips) {#list_gips .anchor-link}
+
+```python
+def list_global_ips(self):
+    orderBy = {"globalIpRecords": {"id": self.DESC}}
+    objectMask = "mask[ipAddress,destinationIpAddress[ipAddress,hardware[id,hostname], virtualGuest[id,hostname]]]"
+    globalIps = self.client.iter_call('SoftLayer_Account', 'getGlobalIpRecords',
+                                      filter=orderBy, mask=objectMask, limit=100)
+    for globalIp in globalIps:
+        ipAddress = globalIp.get('ipAddress', {})
+        associatedHostname = ""
+        # If the global IP is assigned somewhere, set the hostname its assigned to
+        if globalIp.get('destinationIpAddress', None) is not None:
+            if 'hardware' in globalIp['destinationIpAddress'].keys():
+                associatedHostname = globalIp['destinationIpAddress']['hardware']['hostname']
+            elif 'virtualGuest' in globalIp['destinationIpAddress'].keys():
+                associatedHostname = globalIp['destinationIpAddress']['virtualGuest']['hostname']
+            else:
+                associatedHostname = globalIp['destinationIpAddress'].get('ipAddress')
+        if associatedHostname != "":
+            print("{} is assigned to {}".format(ipAddress.get('ipAddress'), associatedHostname))
+        else:
+            print("{} is unassigned".format(ipAddress.get('ipAddress')))
+```
+
+## [Invoices](#invoices) {#invoices .anchor-link}
+
+```python
+
+def invoices(self):
+    """This method will go through ALL invoices on an account.
+
+    The older an account is, the more invoices it will have, so queries like this must be split up. Otherwise
+    you will get API errors or timeouts. Once we have the invoice ID, we can get each invoices details seperately
+    which will greatly speed up this process
+    """
+    orderBy = {"invoices": {"id": self.DESC}}
+
+    objectMask = "mask[id, createDate, statusCode]"
+    invoices = self.client.iter_call("SoftLayer_Account", "getInvoices", filter=orderBy, mask=objectMask)
```

Comment:
add a limit on method, this can produce a time out error in case the user has a much invoices

Reply:
`iter_call` automatically adds a limit, the default is 100